### PR TITLE
Adding support for SQL Server to the base Keycloak Docker

### PR DESCRIPTION
**Pull request by michaeljmcd**
Update on PR #163 and #167  that squashes all commits down to a single pull commit that lives in a new branch. This PR was created as the previous one had been automatically closed.

JIRA request: https://issues.jboss.org/browse/KEYCLOAK-9903

### CODE REVIEW

**Comment by heyjared** [comment]
Any updates on getting this merged?

**Comment by kluiko** [comment]
we need this merge.

File: docker-compose-examples/README.md [hunk 1]
```diff
@@ -30,6 +30,20 @@ Note - This example is not currently working as MySQL is not ready to receive co
...
+
+The `keycloak-mssql.yml` template creates a volume for Microsoft SQL Server and starts Keycloak connected to a Microsoft SQL Server instance. 
+
+Run the example with the following command:
+
+    docker-compose -f keycloak-mssql.yml up
+    
+Note - This example uses an additional container to create the keycloak database prior to loading the keycloak application.  In addition, the keycloak container can be rebuilt using
+    
+    docker-compose -f .\docker-compose-examples\keycloak-mssql.yml build
```
Comment (slaskawi) [inline docker-compose-examples/README.md]: Other documentation bits use Linux type of paths (with `/`). Could you please update it?

File: docker-compose-examples/keycloak-mssql.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            - "8080:8080"
+        environment:
+            - KEYCLOAK_USER=admin
+            - KEYCLOAK_PASSWORD=admin
+            - DB_VENDOR=mssql
+            - DB_USER=sa
+            - DB_PASSWORD=Password!23
+            - DB_ADDR=mssql
+            - DB_DATABASE=Keycloak
+volumes:
```
Comment (slaskawi) [inline docker-compose-examples/keycloak-mssql.yml]: Do we need this volumes? If not, please remove them.

File: server/Dockerfile [hunk 3]
```diff
@@ -4,6 +4,7 @@ ENV KEYCLOAK_VERSION 5.0.0
 ENV JDBC_POSTGRES_VERSION 42.2.5
 ENV JDBC_MYSQL_VERSION 5.1.46
 ENV JDBC_MARIADB_VERSION 2.2.3
+ENV JDBC_MSSQL_VERSION 7.0.0.jre8
```
Comment (slaskawi) [inline server/Dockerfile]: Could you please update it to the latest version? See https://mvnrepository.com/artifact/com.microsoft.sqlserver/mssql-jdbc

File: docker-compose-examples/README.md (hunk 1 above)
Comment (michaeljmcd) [reply docker-compose-examples/README.md]: Absolutely - the latest push switches to forward slashes.

File: docker-compose-examples/keycloak-mssql.yml (hunk 2 above)
Comment (michaeljmcd) [reply docker-compose-examples/keycloak-mssql.yml]: Volumes removed. They were present from the POCs for which this was being used.

File: server/Dockerfile (hunk 3 above)
Comment (michaeljmcd) [reply server/Dockerfile]: Upgraded to the latest JRE8 driver.

**Comment by slaskawi** [comment]
@michaeljmcd We still need a rebase as there are conflicts in your Pull Request. Could you please rebase your changes?

**Comment by michaeljmcd** [comment]
> 
> 
> @michaeljmcd We still need a rebase as there are conflicts in your Pull Request. Could you please rebase your changes?

Absolutely. I just pushed changes to accommodate the rebase.

**Comment by slaskawi** [comment]
@michaeljmcd Everything looks nice! The Docker Compose example works really well. Thanks a lot for your contribution.

@abstractj This one is ready to be merged.

**Pull request merged**